### PR TITLE
Private endpoint to return project developer's projects

### PR DESCRIPTION
**Pull request by agnessa**
This adds a new list endpoint under `api/v1/account/projects`, which only returns projects of the signed in project developer.

It was implemented similarly to the public search, e.g. filtering by full text was reused.

What I was also hoping to reuse was the sorting. However, here I have a doubt - how do we even sort by the columns in the design?
- name - fine (that works)
- category - theoretically easy, but we only store the English slug in the db, so not easy to sort by translation
- status - same problem
- instrument types - almost same problem, but it's an array of English slugs
- value, i.e. ticket size - almost same problem, but in fact we probably want to sort not by translation, but by the amount...
- location - I'm guessing this column is a combination of municipality and country, so theoretically we could just sort by municipality name?

So in the end I did not implement sorting, except for sorting by name which was already working - wasn't sure how to proceed with this.

## Testing instructions

swagger docs

## Tracking

https://vizzuality.atlassian.net/browse/LET-581

### CODE REVIEW

File: backend/spec/requests/api/v1/accounts/projects_spec.rb [hunk 1]
```diff
@@ -62,6 +62,76 @@
...
+      parameter name: "filter[full_text]", in: :query, type: :string, required: false, description: "Filter records by provided text."
+      parameter name: :sorting, in: :query, type: :string, required: false, description: "Sort records.",
+        enum: [
+          "name asc", "name desc",
+          # "category asc", "category desc", # TODO: this is stored as a slug in the db, i.e. in English
+          # "municipality asc", "municipality desc", # TODO: this shows as 'Leticia, Colombia' in the design
+          # "ticket_size asc", "ticket_size desc", # TODO: this is stored as a slug in the db, i.e. in English
+          # "instrument_type asc", "instrument_type desc", # TODO: this is an array of slugs
+          # "status asc", "status desc", # TODO: this is stored as a slug in the db, i.e. in English
+          "created_at asc", "created_at desc"
```
Comment (martintomas) [inline backend/spec/requests/api/v1/accounts/projects_spec.rb:81]: Hmm, I think that to cover all these sorting options with `API::Sorter` is gonna be tricky. Maybe we could rewrite this service to use ransack, which Admin is already using and just add some transformation table which translates FE option to ransack params. Difficult part is that `API::Sorter` is used for sorting everything (PDs/Projects/etc.) so preparation of ransack params would need to take this into account and tweak attributes appropriately --> for example `name` is localized for Projects, but not localized for Accounts, etc.

**Review by martintomas — APPROVED** [review]
Looks good ;). Only difference against public endpoint which I see is missing pagination, which is fine 👍

**Comment by martintomas** [comment]
> there are now separate user stories for sorting and searching of this list, probably low priority

Searching should work  - ofc if we want same full text searching. Sorting is different story, that will need separate user story.

File: backend/spec/requests/api/v1/accounts/projects_spec.rb (hunk 1 above)
Comment (tsubik) [reply backend/spec/requests/api/v1/accounts/projects_spec.rb:81]: Yeah, I think using ransack in or instead of `API::Sorter` could make things easier.

**Comment by tsubik** [comment]
I think it's worth to add pagination, as they will need it for sure.

**Comment by tsubik** [comment]
What about pagination? Separate story or here?

**Comment by agnessa** [comment]
I omitted it, because there don't seem to be pagination links in the design; and it's unlikely those lists are going to be long anyway

**Review by tsubik — APPROVED** [review]

**Pull request merged**